pkg/controller/direct/cloudbuild/workerpool_controller.go

```diff
@@ -135,10 +135,20 @@ func (m *model) AdapterForObject(ctx context.Context, reader client.Reader, u *u
 	if err != nil {
 		return nil, err
 	}
+
+	projectIDAndNum := &refs.ProjectIDAndNum{ID: projectRef.ProjectID}
+	network, found, err := unstructured.NestedString(u.Object, "status", "observedState", "networkConfig", "peeredNetwork")
+	if err == nil || found {
+		tokens := strings.Split(network, "/")
```

Comment:
I don't understand this comment, mind elaborating a little bit more? We use the computeNetworkRef in the mapper to find the compute network's external value. The problem is the GCP server converts the projectID (in request) to projectNum (in response), so the  `spec` contains projectID and the `status` contains projectNum. Then the next reconciliation `compareProto`  always raise error because it detects diff in this immutable field.

Reply:
So IIUC what we're doing here is parsing a reference to a network resource.  That should be parseable using the ComputeNetworkRef functions, because I would hope the format is the same as the format used in an `external` ComputeNetworkRef.

Reply:
Unfortunately, no. The computeNetworkRef does not support parsing a single string yet. We should migrate it to the new reference function format.